compiler/q-implant/script/TorchExtractor.py

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import torch
+import torch.nn
+import torch.quantization
+import numpy as np
+import collections
+import json
+import torch.nn.quantized.modules.linear
+
+
+def quantize_tensor(tensor: torch.Tensor, scale, zero_point,
+                    dtype=np.int8) -> torch.Tensor:
+    if dtype not in (np.uint8, np.int8, np.int32):
+        raise Exception('Please check dtype')
+    new_tensor = tensor.clone().detach().numpy()
+    new_tensor = new_tensor / scale + zero_point
+    return new_tensor.astype(dtype)
+
+
+class TorchExtractor:
+    qdtype_mapping = {
+        torch.quint8: {
+            'str': "uint8",
+            'np': np.uint8
+        },
+        torch.qint8: {
+            'str': "int8",
+            'np': np.int8
+        },
+        torch.qint32: {
+            'str': "int32",
+            'np': np.int32
+        }
+    }
+
+    @staticmethod
+    def permute(tensor: torch.Tensor) -> torch.Tensor:
+        dim = len(tensor.shape)
+        if dim == 4:  # NCHW to NHWC
+            tensor = tensor.permute(0, 2, 3, 1)
+        return tensor
+
+    def __init__(self, quantized_model: torch.nn.Module, json_path: str,
+                 partial_graph_data: None):
+        self.__np_idx = 0
+        self.__input_dtype = None
+        self.__graph_data = collections.OrderedDict()
+        self.__partial_graph_data = partial_graph_data
+        self.__json_path = json_path
+        idx = json_path.rfind(os.path.sep)
+        if idx == -1:
+            self.__dir_path = ""
+            self.__json_file_name = json_path
+        else:
+            self.__dir_path = json_path[:idx + 1]
+            self.__json_file_name = json_path[idx + 1:]
```

Comment:
```suggestion
        self.__dir_path, self.__json_file_name = os.path.split(json_path)
```
I think I can improve your code like this.